samples/TerraFX/Graphics/HelloSmoke.cs

```diff
@@ -0,0 +1,364 @@
+                graphicsContext.Copy(indexBuffer, indexStagingBuffer);
+
+                return indexBuffer;
+            }
+
+            static GraphicsBuffer CreateConstantBuffer(GraphicsContext graphicsContext)
+            {
+                var constantBuffer = graphicsContext.Device.MemoryAllocator.CreateBuffer(GraphicsBufferKind.Constant, GraphicsResourceCpuAccess.Write, 256);
+
+                var pConstantBuffer = constantBuffer.Map<Matrix4x4>();
+                pConstantBuffer[0] = Matrix4x4.Identity;
+                constantBuffer.Unmap(0..sizeof(Matrix4x4));
+
+                return constantBuffer;
+            }
+
+            static GraphicsTexture CreateTexture3D(GraphicsContext graphicsContext, GraphicsBuffer textureStagingBuffer, bool isQuickAndDirty)
+            {
+                const uint TextureWidth = 256;
+                const uint TextureHeight = 256;
+                const ushort TextureDepth = 256;
+                const uint TextureDz = TextureWidth * TextureHeight;
+                const uint TexturePixels = TextureDz * TextureDepth;
+                const uint TextureSize = TexturePixels * 4;
+
+                var texture3D = graphicsContext.Device.MemoryAllocator.CreateTexture(GraphicsTextureKind.ThreeDimensional, GraphicsResourceCpuAccess.None
+                    , TextureWidth, TextureHeight, TextureDepth
+                    , texelFormat: TexelFormat.RGBA4xByte);
+                var pTextureData = textureStagingBuffer.Map<UInt32>();
+
+                var random = new Random(Seed: 1);
+
+                // start with random speckles
+                for (uint n = 0; n < TexturePixels; n++)
+                {
+                    // convert n to indices
+                    float x = n % TextureWidth;
+                    float y = (n % TextureDz) / TextureWidth;
+                    float z = n / TextureDz;
+                    // convert indices to fractions in the range [0, 1)
+                    x /= TextureWidth;
+                    y /= TextureHeight;
+                    z /= TextureHeight;
+                    // make x,z relative to texture center
+                    x -= 0.5f;
+                    z -= 0.5f;
+                    // get radius from center, clamped to 0.5
+                    float radius = MathF.Abs(x); // MathF.Sqrt(x * x + z * z);
+                    if (radius > 0.5f)
+                        radius = 0.5f;
+                    // scale as 1 in center, tapering off to the edge
+                    float scale = 2 * MathF.Abs(0.5f - radius); 
+                    // random value scaled by the above
+                    float rand = (float)random.NextDouble();
+                    if (!isQuickAndDirty &&  rand < 0.99)
+                        rand = 0;
+                    uint value = (byte)(rand * scale * 255);
+                    pTextureData[n] = (uint)(value | value << 8 | value << 16 | value << 24);
+                }
+                if (!isQuickAndDirty)
```

Comment:
newline needed between blocks:
```suggestion

                if (!isQuickAndDirty)
```

Reply:
ok